File: trunk/quote/Projects/PluginModel/QuoteTypeList.vb
Imports System.ComponentModel

Public Class QuoteTypeList : Inherits System.ComponentModel.StringConverter

    Public Shared PRODUCTION As String = "Production"
    Public Shared PILOT As String = "Pilot"
    Public Shared PROVE As String = "Prove"
    Public Shared SINGLE_DEFINATE As String = "Single Definate"

    Public Overloads Overrides Function _
        GetStandardValues(ByVal context As  _
        System.ComponentModel.ITypeDescriptorContext) _
        As TypeConverter.StandardValuesCollection

        Dim l As New List(Of String)
        l.AddRange(New String() {
            PRODUCTION, _
            PILOT, _
            PROVE, _
            SINGLE_DEFINATE
        })
        Return New StandardValuesCollection(l)
    End Function

    Public Overloads Overrides Function _
        GetStandardValuesExclusive(ByVal context _
        As System.ComponentModel.ITypeDescriptorContext) _
        As Boolean

        Return True
    End Function

    Public Overrides Function _
        GetStandardValuesSupported(ByVal context As ITypeDescriptorContext) As Boolean

        Return True
    End Function

End Class

File: trunk/quote/Projects/PluginModel/QuoteTypeList.cs
namespace Model
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Linq;

    public class QuoteTypeList : System.ComponentModel.StringConverter
    {
        public const string Production = "Production";
        
        public const string Pilot = "Pilot";
        
        public const string Prove = "Prove";

        public const string SingleDefinite = "Single Definate";
        
        public override TypeConverter.StandardValuesCollection GetStandardValues(System.ComponentModel.ITypeDescriptorContext context)
        {
            List<string> l = new List<string>();
            l.AddRange(new string[] 
            {
                Production,
                Pilot,
                Prove,
                SingleDefinite
            });
            
            return new StandardValuesCollection(l);
        }

        public override bool GetStandardValuesExclusive(System.ComponentModel.ITypeDescriptorContext context)
        {
            return true;
        }

        public override bool GetStandardValuesSupported(ITypeDescriptorContext context)
        {
            return true;
        }
    }
}
